pkg/client-sdk/wallet.go

```diff
@@ -0,0 +1,273 @@
+package arksdk
+
+import (
+	"bytes"
+	"context"
+	"encoding/hex"
+	"fmt"
+
+	"github.com/ark-network/ark/common/tree"
+	"github.com/btcsuite/btcd/btcec/v2/ecdsa"
+	"github.com/btcsuite/btcd/btcec/v2/schnorr"
+	"github.com/btcsuite/btcd/chaincfg/chainhash"
+	"github.com/btcsuite/btcd/txscript"
+	"github.com/decred/dcrd/dcrec/secp256k1/v4"
+	"github.com/vulpemventures/go-elements/address"
+	"github.com/vulpemventures/go-elements/network"
+	"github.com/vulpemventures/go-elements/payment"
+	"github.com/vulpemventures/go-elements/psetv2"
+	"github.com/vulpemventures/go-elements/transaction"
+)
+
+type Wallet interface {
+	PubKey() *secp256k1.PublicKey
+	PubKeySerializeCompressed() []byte
+	SignPsetForAddress(pset *psetv2.Pset, address string) error
```

Comment:
SignTransactionForAddress would work both on bitcoin and elements.

Reply:
would be better to change this method to:
```suggestion
	SignTransaction(tx string) (string, error)
```

This way we can pass either a psbt, pset or raw transaction and also we extend its definition to HD wallets as expecting to pass a single address might fit well for a single-key but not for other kind of wallets

Reply:
cc @sekulicd